source/position/position_api.cpp

```diff
@@ -0,0 +1,87 @@
+/* Copyright © 2001-2019, Canal TP and/or its affiliates. All rights reserved.
+
+This file is part of Navitia,
+    the software to build cool stuff with public transport.
+
+Hope you'll enjoy and contribute to this project,
+    powered by Canal TP (www.canaltp.fr).
+Help us simplify mobility and open public transport:
+    a non ending quest to the responsive locomotion way of traveling!
+
+LICENCE: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+Stay tuned using
+twitter @navitia
+channel `#navitia` on riot https://riot.im/app/#/room/#navitia:matrix.org
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+
+#include "position_api.h"
+#include "utils/paginate.h"
+
+namespace navitia {
+namespace position {
+
+void vehicle_positions(PbCreator& pb_creator,
+                       const std::string& filter,
+                       size_t count,
+                       int depth,
+                       size_t start_page,
+                       const std::vector<std::string>& forbidden_uris,
+                       const boost::optional<boost::posix_time::ptime>& since,
+                       const boost::optional<boost::posix_time::ptime>& until) {
+    const type::Data& data = *pb_creator.data;
+
+    const auto start = get_optional_value_or(since, pb_creator.now);
+    const auto end = get_optional_value_or(until, pb_creator.now + boost::posix_time::minutes(2 * 10));
```

Comment:
I found use of neg_infin and pos_infin while initializing since and until : https://github.com/CanalTP/navitia/blob/3ab7e2721dd127d30c0e4b6c7a5bfd5d915efbc5/source/disruption/traffic_reports_api.cpp#L366